implementingCustomPolicies.dox

```diff
@@ -39,6 +39,31 @@ In addition, policies keeping references to values **must** swap the held refere
 ### Concurrency Requirements
 Policies do not need to take any steps to ensure thread-safety. The cache will properly protect policies to ensure no concurrent operations occur.
 
+## Constraint Policy
+
+In addition to the shared requirements above, a constraint policy must implement the following methods in order to be valid:
+
+* `bool can_add(const Key& key, const Item<Value>& item)`
+
+    Should return `true` if the item can be inserted in the cache while still satisfying the constraint.
+    @note Returning `true` does not guarantee that the object matching the key will be inserted in cache.
+
+* `bool can_replace(const Key& key, const Item<Value>& old_item, const Item<Value>& new_item)`
+
+    Should return `true` if the value associated with `key` can be changed from `old_item` to `new_item` while still
+    satisfying the constraint.
+    @note Returning `true` does not guarantee that the object matching the key will be updated.
```

Comment:
what happens if we return false ? is the old_item invalidated/evicted from the cache ? :thinking:

Reply:
Nope! If false is returned it means there isn't enough room in the cache to update the object (most likely because the object grew). The cache takes this information and attempts to evict some items to make room (evictions will only actually occur if the new value is deemed preferable to the evicted items by the insertion policy). 

However there is an error in the docs, because returning true from `can_replace` means there _is_ enough room to replace, so it guarantees replacement. I'll fix this